drivers/gpio/gpio_intel_socfpga.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT intel_socfpga_gpio
+
+/**
+ * @file
+ * @brief Intel GPIO SoCFPGA Controller Driver
+ *
+ * The GPIO controller on Intel SoCFPGA serves
+ * as GPIO modules. This driver provides
+ * the GPIO functionality.
+ *
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_intel_socfpga);
+
+/* Check if reset property is defined */
+#define GPIO_INTEL_INST_RESET_EN(inst)	DT_NODE_HAS_PROP(DT_DRV_INST(inst), resets)
+#define GPIO_INTEL_CHECK_RESET(inst)	GPIO_INTEL_INST_RESET_EN(inst) ||
+#define GPIO_INTEL_RESET_SUPPORT	(DT_INST_FOREACH_STATUS_OKAY(GPIO_INTEL_CHECK_RESET) 0)
+
+#if GPIO_INTEL_RESET_SUPPORT
+#include <zephyr/drivers/reset.h>
+#endif
+
+#define DEV_CFG(_dev) ((const struct gpio_intel_socfpga_config *)(_dev)->config)
+
+#define DEV_DATA(_dev) ((struct gpio_intel_socfpga_data *const)(_dev)->data)
+
+
+#define GPIO_SWPORTA_DDR_OFFSET	0x04
+#define GPIO_EXT_PORTA_OFFSET	0x50
+#define GPIO_ID_CODE_OFFSET	0x64
+#define GPIO_VER_ID_CODE_OFFSET	0x6c
+
+#define PMUX_SEL_OFFSET		4
+#define PMUX_40_SEL		40
+#define PMUX_40_SEL_GAP		96
+#define PMUX_GPIO_VAL		8
+
+struct gpio_intel_socfpga_config {
+	/** GPIO common config */
+	struct gpio_driver_config gpio_config;
+	/** GPIO MMIO mapped physical address */
+	DEVICE_MMIO_NAMED_ROM(gpio_mmio);
+	/** GPIO port number (0 or 1) */
+	const uint32_t gpio_port;
+	/** GPIO pin number per port */
+	uint32_t ngpios;
+#if WDT_DW_RESET_SUPPORT
```

Comment:
This seems out of place. Is there a Kconfig value missing? I can't see this symbol being used anywhere in Zephyr currently.

Reply:
Updated to correct GPIO based Macro